### PR TITLE
Refactor and decouple commons from config/context etc

### PR DESCRIPTION
**Pull request by r0h4n**
- Remove redundant samples, code

tendrl-bug-id: Tendrl/commons#90

### CODE REVIEW

File: tendrl/commons/flows/base_flow.py [hunk 1]
```diff
@@ -39,17 +38,10 @@ def __init__(
...
 
+        # Add etcd_client to parameters via child flows
         self.parameters.update({'log': []})
-        etcd_kwargs = {
-            'port': int(self.config.get("commons", "etcd_port")),
-            'host': self.config.get("commons", "etcd_connection")
-        }
-
-        self.etcd_client = etcd.Client(**etcd_kwargs)
-        self.parameters.update({'etcd_client': self.etcd_client})
```
Comment (shtripat) [inline tendrl/commons/flows/base_flow.py]: I remember we had done this for a specific reason where we needed etcd_client in further jobs created while import cluster flow on other nodes. How we are planning to achieve this now?

File: tendrl/commons/manager/rpc_job_process.py [hunk 2]
```diff
@@ -3,26 +3,21 @@
...
-    def __init__(self, syncJobThread):
-        self.config = syncJobThread._manager._config
-        etcd_kwargs = {
-            'port': int(self.config.get("commons", "etcd_port")),
-            'host': self.config.get("commons", "etcd_connection")
-        }
-
-        self.client = etcd.Client(**etcd_kwargs)
-        self.syncJobThread = syncJobThread
+    def __init__(self, syncjobthread, etcd_client):
```
Comment (shtripat) [inline tendrl/commons/manager/rpc_job_process.py]: Below, the creation code for EtcdRpc is not passing etcd_client to the constructor.

File: tendrl/commons/persistence/etcd_persister.py [hunk 3]
```diff
@@ -1,14 +1,8 @@
-from tendrl.commons.etcdobj.etcdobj import Server as etcd_server
 from tendrl.commons.persistence.persister import Persister
 
 
 class EtcdPersister(Persister):
-    def __init(self, config):
-        super(EtcdPersister, self).__init__(config)
```
Comment (shtripat) [inline tendrl/commons/persistence/etcd_persister.py]: I feel passing config was more flexible option and later if we node something more from config, we would have to change the signature of constructor again to pass additional data.

File: tendrl/commons/flows/base_flow.py (hunk 1 above)
Comment (r0h4n) [inline tendrl/commons/flows/base_flow.py]: creating an etcd_client is costly (default creates a httpconnectionpool of 10), hence it is ideal that the rpc layer provides an instance of the etcd_client by appending it to parameters during flow invocation

File: tendrl/commons/manager/rpc_job_process.py (hunk 2 above)
Comment (r0h4n) [inline tendrl/commons/manager/rpc_job_process.py]: Ack, will fix

File: tendrl/commons/persistence/etcd_persister.py (hunk 3 above)
Comment (r0h4n) [inline tendrl/commons/persistence/etcd_persister.py]: The persister and its methods should be atomic in nature, eg: save_volume(volume) should only save the volume to etcd and disk.

The reason for not sending a config to a persister is to keep it sane and simple :)

File: tendrl/commons/persistence/etcd_persister.py [hunk 4]
```diff
@@ -1,14 +1,8 @@
...
-    def get_store(self):
-        etcd_kwargs = {
-            'port': int(self._config.get("commons", "etcd_port")),
-            'host': self._config.get("commons", "etcd_connection")
-        }
-        return etcd_server(etcd_kwargs=etcd_kwargs)
+    def __init(self, etcd_client):
+        super(EtcdPersister, self).__init__()
+        # Child classes responsible to provide valid etcd client
+        self.etcd_client = etcd_client
```
Comment (r0h4n) [inline tendrl/commons/persistence/etcd_persister.py]: This should also be injected by the child class, will remove

File: tendrl/commons/flows/base_flow.py (hunk 1 above)
Comment (shtripat) [inline tendrl/commons/flows/base_flow.py]: ok

**Pull request merged**